### DISCUSSION TITLE
Auto Offset based on what's on the screen

### DISCUSSION THREAD

**Discussion by cdolek** [rxhanson/Rectangle#308] [Ideas]
I'm using iTerm docked to the bottom of my screen, and want the other windows (i.e. VSCode, Sublime Text etc.) to be placed side by side at the top but I want to be able to see iTerm at the bottom as well. 

With Spectacle and now Rectangle, I had to align my top windows (Left Half | Right Half) first and then adjust their heights.

     /--------------------|--------------------\
     |                    |                    |
     |       win1         |        win2        |
     |                    |                    |
     |--------------------|--------------------|
     |                  iTerm                  |     
     \-----------------------------------------/

Could the actions using screen height detect if there is a window covering the bottom (or top) of the screen and size the window based on the remaining max height?

Thanks

**Comment by rxhanson** [comment]
This is something that I've thought about a fair amount. It's actually a decently complicated problem to solve in a generic sense, where there might be more windows open than just those few, and factoring in thresholds for what size window is too small or what to do if a window doesn't fit quite right in the available space. I don't have plans to implement this in Rectangle, but would consider a pull request for it.

A while back, I did implement a first cut at similar behavior in [Hookshot](https://hookshot.app), with the "Fill Left" and "Fill Right" actions.  These actions only take into account the window immediately above or below the window being moved, will always fill the full height of the screen, and will perform a left or right half action if the window doesn't quite fit right. I have been planning to improve those actions and add actions that would solve your scenario, but I just haven't had a chance to get to it.
